src/controllers/Interactions.ts

```diff
@@ -0,0 +1,103 @@
+import { Client, Interaction, TextChannel } from "discord.js";
+import { DateTime } from "luxon";
+import LeaderboardRepository from "../repositories/LeaderboardRepository";
+import QueueRepository from "../repositories/QueueRepository";
+import { BallChaser } from "../types/common";
+import getDiscordChannelById from "../utils/getDiscordChannelById";
+import getEnvVariable from "../utils/getEnvVariable";
+import MessageBuilder from "../repositories/helpers/MessageBuilder";
+
+const NormClient = new Client({ intents: "GUILDS" });
+const queueChannelId = getEnvVariable("queue_channel_id");
+
+
+export async function buttonEmbeds(queueChannel: TextChannel): Promise<void> {
+  
+  const ballchasers = await QueueRepository.getAllBallChasersInQueue();
+  if (ballchasers == null) {
+
+    await queueChannel.send({ 
+      components: [MessageBuilder.queueButtons],
+      embeds: [MessageBuilder.emptyQueueMessage()] });
+
+  } else {
+
+    await queueChannel.send({ 
+      components: [MessageBuilder.queueButtons],
+      embeds: [MessageBuilder.activeQueueMessage(ballchasers)] });
+  }
+}
+
+NormClient.on("interactionCreate", async (buttonInteraction: Interaction) => {
+
+  if (!buttonInteraction.isButton()) return;
+  await buttonInteraction.deferReply();
+
+  switch (buttonInteraction.customId) {
+    case "joinQueue": {
+
+      const queueMember = await QueueRepository.getBallChaserInQueue(buttonInteraction.user.toString());
+      const leaderboardMember = await LeaderboardRepository.getPlayerStats(buttonInteraction.user.toString());
+
+      if (!queueMember) {
+        const player: BallChaser = {
+          id: buttonInteraction.user.toString(),
+          isCap: false,
+          mmr: leaderboardMember ? leaderboardMember.mmr : 100,
+          name: buttonInteraction.user.username,
+          queueTime: DateTime.now(),
```

Comment:
I think for `queueTime` here is where we want to add the 60 minutes. That way when we implement the stale queue timer it's a simple check.

Reply:
@N0ise9 If you want to fix that as I'm not quite sure how the current bot code works in this instance.

Reply:
Resolved.